### PR TITLE
images: bundle containers

### LINKED ISSUE #457: Include containers necessary for initial deployment inside Aquarium image
When deploying Ceph on each node[1], the ceph container image is pulled from (at the time of writing) docker.io/ceph/ceph:v16. That takes a certain amount of time (and also an internet connection).

Once #449 is merged, we'll also be pulling quay.io/coreos/etcd:latest, and that presumably needs to complete before Aquarium can actually be used at all.

I assume there may be other containers in future too.

So, I had a crazy idea. Can we do a `podman pull` of all necessary containers at image build time, so that they are present and ready to use immediately inside Aquarium?

[1] Actually, it looks like the ceph container is pulled automagically at runtime _before_ deployment, I assume as a result of Aquarium periodically running `cephadm gather-facts` and/or `cephadm inventory`, but the point still stands, this requires a certain amount of time, and an internet connection.

### PR DESCRIPTION
**Pull request by jecluis**
Resolves #457 

Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

**Review by jhesketh — COMMENTED** [review]
Unless I'm missing something, this script isn't actually ran, so we're not including the images so much as a helper for them.

**Comment by jecluis** [comment]
On Mon, May 3, 2021, 04:46 Joshua Hesketh ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> Unless I'm missing something, this script isn't actually ran, so we're not
> including the images so much as a helper for them.
>
Much like config.sh, disk.sh is run at a given point in the kiwi workflow.
In this case, disk.sh is called on the final image instead of the working
filesystem tree.

So no, it's not a helper, and it is really including the images in the
build.

**Comment by jhesketh** [comment]
> On Mon, May 3, 2021, 04:46 Joshua Hesketh ***@***.***> wrote: ***@***.**** commented on this pull request. Unless I'm missing something, this script isn't actually ran, so we're not including the images so much as a helper for them.
> Much like config.sh, disk.sh is run at a given point in the kiwi workflow. In this case, disk.sh is called on the final image instead of the working filesystem tree. So no, it's not a helper, and it is really including the images in the build.

Ah, okay, I had wondered if something like that was the case. (I'm not very familiar with kiwi). :+1:

File: images/microOS/disk.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+echo "nameserver 1.1.1.1" > /etc/resolv.conf
+
+# and we need a basic running system too
+mount -t sysfs none /sys
+mount -t devtmpfs none /dev
+mount -t tmpfs none /dev/shm
+mount -t proc none /proc
+mount -t tmpfs none /run
+
+/usr/bin/podman pull quay.io/coreos/etcd:latest
```
Comment (tserong) [inline images/microOS/disk.sh:40]:
```suggestion
# setting "--events-backend none" means podman doesn't try
# (and fail) to log a "system refresh" event to the journal
/usr/bin/podman --events-backend none pull quay.io/coreos/etcd:latest
```

File: images/microOS/disk.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+# we don't get to use cephadm directly because it will
+# try running a container inside the chroot, and that
+# fails with a bang.
+/usr/bin/podman pull docker.io/ceph/ceph:v16
```
Comment (tserong) [inline images/microOS/disk.sh:44]:
```suggestion
/usr/bin/podman --events-backend none pull docker.io/ceph/ceph:v16
```

**Review by tserong — APPROVED** [review]
Works like a charm, although I noticed this ugly thing in the middle of the disk.sh output:

```
time="2021-05-03T07:39:44Z" level=error msg="unable to write system event: \"write unixgram @00148->/run/systemd/journal/socket: sendmsg: no such file or directory\""
```

Turns out podman tries to log a "system refresh" event to the journal. In the context of this script, I don't think we care at all about podman events, so I've suggested adding `--events-backend none` to each of the `podman pull` invocations to get rid of this.

--events-backend none

**Review by tserong — APPROVED** [review]

**Pull request merged**